drivers/charger/sbs_charger/sbs_charger.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2023 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sbs_sbs_charger
+
+#include "sbs_charger.h"
+
+#include <zephyr/drivers/charger.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+LOG_MODULE_REGISTER(sbs_charger);
+
+static int sbs_cmd_reg_read(const struct device *dev, uint8_t reg_addr, uint16_t *val)
+{
+	const struct sbs_charger_config *cfg;
+	uint8_t i2c_data[2];
+	int status;
+
+	cfg = dev->config;
+	status = i2c_burst_read_dt(&cfg->i2c, reg_addr, i2c_data, ARRAY_SIZE(i2c_data));
+	if (status < 0) {
+		LOG_ERR("Unable to read register");
+		return status;
+	}
+
+	*val = sys_get_le16(i2c_data);
+
+	return 0;
+}
+
+static int sbs_charger_get_prop(const struct device *dev, charger_property_t prop,
+				struct charger_value *val)
+{
+	uint16_t reg_val = 0;
+	int ret;
+
+	switch (prop) {
+	case CHARGER_PROP_ONLINE:
+		ret = sbs_cmd_reg_read(dev, SBS_CHARGER_REG_STATUS, &reg_val);
+		if (reg_val & SBS_CHARGER_STATUS_AC_PRESENT) {
+			val->value.online = CHARGER_ONLINE_FIXED;
```

Comment:
Why isn’t this using a variant type set/reset function?

Reply:
I don't understand what you mean. 
Electrically, we are reading a bit that indicates the presence of a travel adaptor to get a state of whether or not the supply is offline, online, or online and programmable (a la USB PPS).

Reply:
Drivers or clients should make use of the getter/setter helper functions, so that we can keep the implementation details internal, ie, use `charger_value_reset_xxx(val, CHARGER_ONLINE_FIXED)`.

Reply:
Hey @gmarull, sorry I'm still a little confused here.

The comment you're leaving is in the driver implementation, so I don't see how we're not keeping the implementation details internal.

Reply:
How the union is defined (ie, how the generic type is implemented) should be hidden from drivers/clients. That leads to much readable code, and less prone to breaking changes.